multinet/db.py

```diff
@@ -301,6 +323,31 @@ def workspace_table_keys(
     return keys
 
 
+# TODO: Ensure current user has writer role
+def create_workspace_table_from_aql(workspace: str, name: str, aql: str) -> str:
+    """Create a new workspace from an aql query."""
+    db = get_workspace_db(workspace, readonly=True)
+
+    if db.has_collection(name):
+        raise AlreadyExists("table", name)
+
+    # In the future, the result of this validation can be
+    # used to determine dependencies in virtual tables
+    try:
+        db.aql.validate(aql)
+        rows = list(db.aql.execute(aql))
```

Comment:
Can we get away with not coercing this to a list here? Database queries are, in general, supposed to be able to handle lots of data and return lots of results.

Reply:
We use it in both `validate_csv` and `coll.insert_many` below. If we were not to coerce it, I believe the cursor would be exhausted by the time we tried to insert the documents into the collection.

Reply:
Ok let's leave it as is for now. I have opened #408 to address it at some point.